src/main/java/com/google/sps/data/Post.java

```diff
@@ -114,18 +123,36 @@ public String toString() {
 
     static class PostComparator implements Comparator<Post> {
         @Override
-        //Orders the most recently posted first 
-        //Reverse timestamp ordering 
         public int compare(Post a, Post b) {
             long currTime = System.currentTimeMillis();
 
             //Cast the timestamps into double between 0 and 1 
-            double aRecency = (double) a.getTimestamp() / (double) currTime;
-            double bRecency = (double) b.getTimestamp() / (double) currTime;
-
-            //Double compare function returns 1 if bRecency is larger than aRecency 
-            //Returns 0 if they're equal and -1 if aRecency is larger 
-            return Double.compare(bRecency, aRecency);
+            double aScore = (double) a.getTimestamp() / (double) currTime;
+            double bScore = (double) b.getTimestamp() / (double) currTime;
+
+            long reactionCountDifference = Math.abs(a.totalReactionCount() - b.totalReactionCount());
+            double reactionBoost = -1.0;
+            for (int i = 0; i < Constants.REACTION_COUNT_BUCKETS.length; i++) {
+                if (reactionCountDifference <= Constants.REACTION_COUNT_BUCKETS[i]) {
+                    reactionBoost = Constants.REACTION_BUCKET_BOOSTS[i]; 
+                    break; 
+                }
+            } 
+            if (reactionBoost == -1.0) {
```

Comment:
dont use == to compare doubles. 

https://stackoverflow.com/questions/41973807/what-is-good-way-to-check-equality-between-2-doubles-in-java/41973846 outlines various options.